onnxruntime/core/providers/cpu/tensor/utils.h

```diff
@@ -192,17 +193,18 @@ struct SliceIteratorBase {
                 dims.size() == extents_.size() &&
                 dims.size() >= steps.size());
 
-    size_t pitch = 1;
+    SafeInt<size_t> pitch = 1;
     // Initial skip, so that input_ points to the first element to copy
     for (size_t i = dims.size(); i-- > 0;) {
       input_ += pitch * starts[i] * element_size_;
-      pitch *= dims[i];
+      pitch *= static_cast<size_t>(dims[i]);
```

Comment:
should this be gsl::narrow to do the check after static cast ? Or do you think it is safe to cast without the check ?

Reply:
Yes we should, but this file is used in CUDA cu files, and they don't support exception. So static_cast is the best candidate.

Reply:
See: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#exception-handling "Exception specification is not supported for __global__ functions."

Reply:
And this file is used in some cu files, but tokenizer.cc is purely for CPU.